go/test/endtoend/vtgate/transaction/trxn_mode_test.go

```diff
@@ -228,3 +232,18 @@ func TestTransactionModes(t *testing.T) {
 	want = `[]`
 	assert.Equal(t, want, got)
 }
+
+func TestTransactionsInStreamingMode(t *testing.T) {
+	defer cluster.PanicHandler(t)
+	ctx := context.Background()
+	conn, err := mysql.Connect(ctx, &vtParams)
+	require.NoError(t, err)
+	defer conn.Close()
+
+	exec(t, conn, "set workload = olap")
+	exec(t, conn, "begin")
+	exec(t, conn, "insert into twopc_user(user_id, name) values(1,'john')")
+	utils.AssertMatches(t, conn, "select * from twopc_user", ``)
```

Comment:
is it normal to expect 0 results if we have inserted a row before?

Reply:
if only we had a test that asserted that we are doing the right thing here... ;)

Reply:
oh, I missed the `exec commit`. We can add another AssertMatches after `commit` to ensure we have our row